# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  b95f7e3cd5cc7534cfc60f0bdbaf3d5267ccd263
Branding: Revert Services.appinfo.name back to 'Thunderbird' after MOZ_APP_BASENAME changed to 'Betterbird'.

diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -4195,8 +4195,17 @@ int XREMain::XRE_mainInit(bool* aExitFla
   if (!mAppData->name) {
     Output(true, "Error: App:Name not specified in application.ini\n");
     return 1;
   }
+
+  // Here is the story:
+  // From 140 we're setting MOZ_APP_BASENAME to "Betterbird". That gets written into application.ini.
+  // With "Betterbird" in `mAppData->name` we run into a lot of trouble, since this is returned by
+  // Services.appinfo.name. This in turn is used here:
+  // https://searchfox.org/comm-central/rev/bdb959086c22039dbcdd97e8fb8911b7331c8ebc/mozilla/toolkit/components/urlformatter/URLFormatter.sys.mjs#109
+  // to format %APP% in any URL, like the one from pref extensions.getAddons.search.browseURL which has value:
+  // https://addons.thunderbird.net/%LOCALE%/%APP%/search/?q=%TERMS%&appver=%VERSION%&platform=%OS%
+  mAppData->name = "Thunderbird";
   if (!mAppData->buildID) {
     Output(true, "Error: App:BuildID not specified in application.ini\n");
     return 1;
   }
